### PR TITLE
opt: fast-path for PageId::encode

### PR DESCRIPTION
**Pull request by rphmeier**
We spend a lot of time encoding Page IDs. 

Most Page IDs (any which are above a 64-byte collision) are short.

### CODE REVIEW

**Comment by rphmeier** [comment]
* **#547** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/547?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#546** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/546?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: core/src/page_id.rs [hunk 1]
```diff
@@ -128,13 +128,27 @@ impl PageId {
 
     /// Encode this page ID to its disambiguated (fixed-width) representation.
     pub fn encode(&self) -> [u8; 32] {
-        let mut uint = Uint::<256, 4>::from(0);
-        for limb in &self.path {
-            uint += Uint::from(limb + 1);
-            uint <<= 6;
-        }
+        if self.path.len() < 10 {
+            // 7 + 6*9 = 61 - max bit-width of a page with depth 9.
```
Comment (pepyakin) [inline core/src/page_id.rs:132]: It's not immediatelly obvious to me, where does this 7 come from?

**Review by pepyakin — APPROVED** [review]

File: core/src/page_id.rs (hunk 1 above)
Comment (rphmeier) [reply core/src/page_id.rs:132]: We do up to 9 shifts of a bit-pattern of 7 bits. The 7-bit pattern is `1000000` and is what you get when you add `1` to `111111`.

**Comment by pepyakin** [comment]
### Merge activity

* **Nov 5, 1:49 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/546).

**Pull request merged**